app/producers/application_producer.rb

```diff
@@ -12,6 +14,12 @@ class ApplicationProducer < Kafka::Client
   #   TOPIC = 'platform.payload-status'
 
   class << self
+    def ping
+      # The partition count method fails if the connection is not alive, so we are
+      # sending a random topic name to it for status checks.
+      kafka.partition_count(Settings.kafka_producer_topics.to_h.values.compact.sample)
```

Comment:
Where do we get the `Settings.kafka_producer_topics` from?
Do we need to rehash it?

Reply:
`Settings.kafka.topics` is indeed convertible to a Hash. See my latest [push](https://github.com/RedHatInsights/compliance-backend/pull/1752/files#diff-8ccd2dde6a3d162300523af2c9d5fb197f1c707042fcc63072e20b750af10f5aR20).